web/src/app/components/GameInput.tsx

```diff
@@ -0,0 +1,108 @@
+import { useRef, useState } from "react";
+
+interface GameInputProps {
+    gameText: string
+    playerFinished: boolean
+    gameStarted: boolean
+    onCorrectKeystroke: () => void;
+    onCompleteWord: () => void;
+}
+
+export default function GameInput({gameText, playerFinished, gameStarted, onCorrectKeystroke, onCompleteWord}: GameInputProps) {
+
+    const [inputString, setInputString] = useState('');
+    const [completedText, setCompletedText] = useState('');
+    const [wordsCompleted, setWordsCompleted] = useState(0);
+    const [isTyping, setIsTyping] = useState(false); 
+
+    const typingTimeout = useRef(0);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const words = gameText.trim().split(/(?<=\s)/) //split and keep whitespace characters
+    const currentWord = words[wordsCompleted] ?? "";
+   
+    if (inputRef.current && gameStarted) {
+        inputRef.current.focus(); 
+    }
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        //disallow arrow key navigation in input field
+        if(e.key === "ArrowLeft" || e.key === "ArrowRight") {
```

Comment:
I think we should add all the arrow keys here, so up and down just in case that someone accidentally presses it.

Reply:
Actually just tested this since text cursor can only go from left to right, up and down won't affect this.